llvm/test/Transforms/FunctionSpecialization/phi-nodes-non-constfoldable.ll

```diff
@@ -0,0 +1,54 @@
+; RUN: opt -passes="ipsccp<func-spec>" -funcspec-min-function-size=20 -funcspec-for-literal-constant -S < %s
```

Comment:
Have you checked `-funcspec-min-function-size=20` is the "size" of bar? The function seems smaller than the previous example just above. I would expect a lower value would allow specialization. If you have checked that the test is passing disregard this comment.